src/Listener.cpp

```diff
@@ -83,6 +85,60 @@ CConfig CTCPListener::ToConfig() const {
     return listenerConfig;
 }
 
+CUnixListener::~CUnixListener() {
+}
+
+bool CUnixListener::Listen() {
+    CString sName = "unix:" + m_sPath;
+    int fd;
+    struct sockaddr_un addr;
+
+    fd = socket(AF_UNIX, SOCK_STREAM, 0);
+    if (fd < 0)
+        return false;
+
+    memset(&addr, 0, sizeof(addr));
+    addr.sun_family = AF_UNIX;
+    strncpy(addr.sun_path, m_sPath.c_str(), sizeof(addr.sun_path)-1);
+
+    CFile::Delete(m_sPath);
+    if (bind(fd, reinterpret_cast<struct sockaddr*>(&addr), sizeof(addr)) < 0) {
+        close(fd);
+        return false;
+    }
+    if (listen(fd, 42) < 0) {
```

Comment:
This loses the error message, which is only at errno at this point, and gets overwritten by close()